File: packages/engine/src/cpus/6502/instructions/sec.js
import { CycleType } from '..'

/**
 * Set Carry
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       SEC           $38  1   2
 */
export function* sec() {
  this.carry = true
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/sec.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Set Carry
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       SEC           $38  1   2

pub fn sec(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.c = true;
}
